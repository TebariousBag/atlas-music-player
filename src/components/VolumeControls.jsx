import { Volume2 } from "lucide-react";

export default function VolumeControls() {
  return (
    <div className="my-4 flex w-full items-center">
      <Volume2 className="fill-french-blue stroke-french-blue mx-2" />
      <input
        type="range"
        min="0"
        max="100"
        className="slider accent-french-blue w-full cursor-pointer"
      />
    </div>
  );
}
