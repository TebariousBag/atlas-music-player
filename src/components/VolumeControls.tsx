import { Volume2 } from "lucide-react";
import { VolumeControlsProps } from "../types";

export default function VolumeControls({
  volume,
  onVolumeChange,
}: VolumeControlsProps) {
  return (
    <div className="my-4 flex w-full items-center">
      <Volume2 className="fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300" />
      <input
        type="range"
        min="0"
        max="100"
        value={volume}
        onChange={(e) => onVolumeChange(Number(e.target.value))}
        className="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
      />
    </div>
  );
}
