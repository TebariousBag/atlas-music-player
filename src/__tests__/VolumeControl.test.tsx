import { render } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import VolumeControls from "../components/VolumeControls";

describe("VolumeControls", () => {
  const mockOnVolumeChange = () => {};

  it("renders with minimum volume (0)", () => {
    const { container } = render(
      <VolumeControls volume={0} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with maximum volume (100)", () => {
    const { container } = render(
      <VolumeControls volume={100} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with medium volume (50)", () => {
    const { container } = render(
      <VolumeControls volume={50} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with low volume (25)", () => {
    const { container } = render(
      <VolumeControls volume={25} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with high volume (75)", () => {
    const { container } = render(
      <VolumeControls volume={75} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders with custom volume (33)", () => {
    const { container } = render(
      <VolumeControls volume={33} onVolumeChange={mockOnVolumeChange} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
