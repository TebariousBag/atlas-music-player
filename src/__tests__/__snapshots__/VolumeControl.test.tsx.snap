// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls > renders with custom volume (33) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="33"
  />
</div>
`;

exports[`VolumeControls > renders with high volume (75) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="75"
  />
</div>
`;

exports[`VolumeControls > renders with low volume (25) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="25"
  />
</div>
`;

exports[`VolumeControls > renders with maximum volume (100) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="100"
  />
</div>
`;

exports[`VolumeControls > renders with medium volume (50) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="50"
  />
</div>
`;

exports[`VolumeControls > renders with minimum volume (0) 1`] = `
<div
  class="my-4 flex w-full items-center"
>
  <svg
    aria-hidden="true"
    class="lucide lucide-volume2 lucide-volume-2 fill-french-blue stroke-french-blue mx-2 dark:fill-gray-300 dark:stroke-gray-300"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 4.702a.705.705 0 0 0-1.203-.498L6.413 7.587A1.4 1.4 0 0 1 5.416 8H3a1 1 0 0 0-1 1v6a1 1 0 0 0 1 1h2.416a1.4 1.4 0 0 1 .997.413l3.383 3.384A.705.705 0 0 0 11 19.298z"
    />
    <path
      d="M16 9a5 5 0 0 1 0 6"
    />
    <path
      d="M19.364 18.364a9 9 0 0 0 0-12.728"
    />
  </svg>
  <input
    class="slider accent-french-blue w-full cursor-pointer dark:accent-gray-300"
    max="100"
    min="0"
    type="range"
    value="0"
  />
</div>
`;
